### ISSUE TITLE
Support redis_skip_client_commands

### ISSUE THREAD

**Issue by cod-y** [bitnami/bitnami-docker-discourse#116]
I'd like to request support for the [DISCOURSE_REDIS_SKIP_CLIENT_COMMANDS](https://review.discourse.org/t/feature-add-setting-to-bypass-sending-redis-client-commands/966) option in Discourse. Setting this as an environment variable and setting the value to "true" should prevent any of the unnecessary `CLIENT` commands to be run in Redis. 

This is required for managed cloud services like GCP Memorystore. See [Memorystore blocked commands](https://cloud.google.com/memorystore/docs/reference/redis-configs#blocked) for more info.

Right now I have something like the following docker-compose.yml

```
version: '2'
services:
  discourse:
    image: 'bitnami/discourse:2'
    volumes:
      - 'discourse_data:/bitnami'
    environment:
      - DISCOURSE_HOSTNAME=www.example.com
      - DISCOURSE_PASSWORD=password
      - DISCOURSE_PORT_NUMBER=5000
      - POSTGRESQL_HOST=postgres-host
      - POSTGRESQL_ROOT_USER=username
      - POSTGRESQL_ROOT_PASSWORD=password
      - POSTGRESQL_NAME=discourse_db
      - DISCOURSE_POSTGRESQL_USERNAME=username
      - DISCOURSE_POSTGRESQL_PASSWORD=password
      - REDIS_HOST=redis-host
      - DISCOURSE_REDIS_SKIP_CLIENT_COMMANDS=true
volumes:
  discourse_data:
    driver: local
```

I'm getting this in the output: https://gist.github.com/cod-y/ebd299ab800715c23da05a72b3d349d5.

Support for this setting would be ideal, but I'd be fine with a simple patch of some sort to get this configuration to apply to Discourse.

**Comment by andresbono** [comment]
Hi @cod-y, you can try to add that config option into the main configuration file. In order to do so, you can try the following:

Modify the `rootfs/init.sh` file to include a line that sets that property:

https://github.com/bitnami/bitnami-docker-discourse/blob/3a2fbffb402947f9b210b87e15d18e93e74e934a/2/debian-9/rootfs/init.sh#L1-L4

```diff
diff --git a/2/debian-9/rootfs/init.sh b/2/debian-9/rootfs/init.sh
index 62e11ce..6545e01 100644
--- a/2/debian-9/rootfs/init.sh
+++ b/2/debian-9/rootfs/init.sh
@@ -1,5 +1,7 @@
 #!/bin/bash

+echo 'redis_skip_client_commands = true' >> /opt/bitnami/discourse/config/discourse.conf
+
 for CONF_FILE in discourse.conf database.yml; do
     if [[ -f "/bitnami/discourse/conf/$CONF_FILE" ]]; then
         warn "A persisted configuration file '$CONF_FILE' was found inside '/bitnami/discourse/conf'.  This feature is not supported anymore. Please mount the configuration file at '/opt/bitnami/discourse/conf' instead."
```

Mount the init file in the container:

```diff
diff --git a/2/debian-9/docker-compose.yml b/2/debian-9/docker-compose.yml
index 1acf76d..3bcefd4 100644
--- a/2/debian-9/docker-compose.yml
+++ b/2/debian-9/docker-compose.yml
@@ -21,6 +21,7 @@ services:
       - redis
     volumes:
       - 'discourse_data:/bitnami'
+      - './rootfs/init.sh:/init.sh'
     environment:
       - POSTGRESQL_HOST=postgresql
       - POSTGRESQL_ROOT_USER=postgres
```

Remove the volumes and do a full restart of the application.

Let us know if that works for you and we will try to implement the environment variable.

**Comment by cod-y** [comment]
Hey @andresbono thank you so much for providing a workaround. I can confirm that this workaround allows me to get it working with cloudsql. Looking forward to official support in the image. I started trying to dig in to see if I could submit a PR for this capability but I got lost 😆. Fwiw I'm happy to give it a try if a maintainer could provide some guidance on slack or something.

**Comment by andresbono** [comment]
Thanks for confirming that it works! We will let you know when the feature is implemented in the image so you can test it again.

**Comment by marcosbc** [comment]
Hi, we just released a new container version, [2.7.0-debian-10-r4](https://github.com/bitnami/bitnami-docker-discourse#270-debian-10-r4). This version uses initialization scripts located in the [`rootfs/` directory](https://github.com/bitnami/bitnami-docker-discourse/blob/master/2/debian-10/rootfs/).

As part of the major refactor of the existing logic, we added the following environment variable which will allow you to solve this issue:

- `DISCOURSE_EXTRA_CONF_CONTENT`: Extra configuration to append to the `discourse.conf` configuration file. No defaults.
